### PR TITLE
fix: remove auto github fethced pages and http-fetcher content from e…

### PR DESCRIPTION
**Pull request by kamiiiel**
see: https://gravitee.atlassian.net/browse/GKO-585

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiExportServiceImpl.java [hunk 1]
```diff
@@ -180,6 +187,34 @@ private void mapCRDMembers(ExecutionContext executionContext, ArrayNode members)
...
+            if (
+                (
+                    PageType.MARKDOWN.name().equals(page.getType()) ||
+                    PageType.SWAGGER.name().equals(page.getType()) ||
+                    PageType.FOLDER.name().equals(page.getType())
+                ) &&
+                page.getSource() != null &&
+                "github-fetcher".equals(page.getSource().getType())
+            ) {
+                iterator.remove();
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiExportServiceImpl.java:214]: If you remove the whole page, you will loose the fetcher configuration

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiExportServiceImpl.java [hunk 2]
```diff
@@ -180,6 +187,34 @@ private void mapCRDMembers(ExecutionContext executionContext, ArrayNode members)
...
+            } else if (
+                PageType.SWAGGER.name().equals(page.getType()) &&
+                page.getSource() != null &&
+                "http-fetcher".equals(page.getSource().getType())
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiExportServiceImpl.java]: Are you sure you cannot have a MARKDOWN with http-fetcher ?

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiExportServiceImpl.java [hunk 3]
```diff
@@ -180,6 +187,34 @@ private void mapCRDMembers(ExecutionContext executionContext, ArrayNode members)
...
+            ) {
+                ((ObjectNode) pageJsonNode.getValue()).remove("content");
+            } else if (PageType.SYSTEM_FOLDER.name().equals(page.getType())) {
+                iterator.remove();
```
Comment (phiz71) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiExportServiceImpl.java]: If you remove the system folder from an API, you will loose the LINK pages that are used in the developer portal to display custom links

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiExportServiceImpl.java (hunk 2 above)
Comment (kamiiiel) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiExportServiceImpl.java]: I added the MARKDOWN as well

**Review by a-cordier — CHANGES_REQUESTED** [review]
Can we reword the commit ? there is a typo in `fetcher`

Also, as discussed with APIM, we will wait fist 4.5.x maintenance release to ship this as an enhancement

**Review by a-cordier — APPROVED** [review]

**Pull request merged**